sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/PruningSuite.scala

```diff
@@ -141,6 +141,14 @@ class PruningSuite extends HiveComparisonTest with BeforeAndAfter {
       Seq("2008-04-08", "11"),
       Seq("2008-04-09", "11")))
 
+  createPruningTest("Partition pruning - with nondeterministic fields",
+    "SELECT value, hr FROM srcpart1 WHERE ds = '2008-04-08' AND rand(7) < 1",
```

Comment:
I'm not sure if it's safe to push it down. For non-deterministic expressions, the order(or number) of  input rows matters. If we push down the deterministic part of filter condition, then the input rows to the remaining filter condition will change and may result to wrong answer.